## Dataframe construction

In [117]:
import pandas as pd

In [122]:
import pandas as pd
df1 = pd.read_csv('../Part 2 Capstone/df2.csv')


/Users/ugp/anaconda2/lib/python2.7/site-packages/IPython/core/interactiveshell.py:2723: DtypeWarning: Columns (16,21,22) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [119]:
for column in df1.columns:
    print df1[column].describe()

count    994039.000000
mean      28342.562048
std       17292.874500
min           0.000000
25%       13806.000000
50%       27612.000000
75%       41418.000000
max       79091.000000
Name: Unnamed: 0, dtype: float64
count    994039.000000
mean     200515.128781
std         514.703215
min      199601.000000
25%      200103.000000
50%      200512.000000
75%      201003.000000
max      201310.000000
Name: BEGIN_YEARMONTH, dtype: float64
count    994039.000000
mean         14.891068
std           9.178259
min           1.000000
25%           7.000000
50%          15.000000
75%          23.000000
max          31.000000
Name: BEGIN_DAY, dtype: float64
count    994039.000000
mean       1290.468152
std         682.994668
min           0.000000
25%         800.000000
50%        1500.000000
75%        1812.000000
max        2359.000000
Name: BEGIN_TIME, dtype: float64
count    994039.000000
mean     200515.131411
std         514.703310
min      199601.000000
25%      200103.000000
50%      2005

In [123]:
df1.shape

(994039, 31)

In [124]:
df1.EVENT_TYPE.unique()

array(['Winter Storm', 'Flood', 'High Wind', 'Tornado', 'Lightning',
       'Waterspout', 'OTHER', 'Heavy Rain', 'Drought', 'Rip Current',
       'Wildfire', 'Tropical Storm', 'Hurricane', 'Tsunami', 'Landslide'], dtype=object)



## Below i will construct a number of dataframes that I will subsequently run both regression and classification models on.




In [125]:
df1.columns

Index([u'Unnamed: 0', u'BEGIN_DAY', u'EPISODE_ID', u'EVENT_ID', u'STATE',
       u'YEAR', u'MONTH_NAME', u'EVENT_TYPE', u'CZ_NAME', u'WFO',
       u'INJURIES_DIRECT', u'INJURIES_INDIRECT', u'DEATHS_DIRECT',
       u'DEATHS_INDIRECT', u'DAMAGE_PROPERTY', u'DAMAGE_CROPS', u'FLOOD_CAUSE',
       u'CATEGORY', u'TOR_F_SCALE', u'TOR_LENGTH', u'TOR_WIDTH',
       u'TOR_OTHER_WFO', u'TOR_OTHER_CZ_STATE', u'END_LAT',
       u'EPISODE_NARRATIVE', u'EVENT_NARRATIVE', u'CASUALTIES',
       u'TOTAL_DEATHS', u'TOTAL_INJURIES', u'DATE', u'COMBINED_DAMAGE'],
      dtype='object')

In [174]:
a = pd.DataFrame(df1.groupby('EVENT_TYPE').CASUALTIES.agg(['count','min', 'max', 'mean', 'sum']).sort_values('sum', axis=0, ascending=False).reset_index().head(11))

In [175]:
a.head(15)

EVENT_TYPE   count  min   max      mean    sum
0        Tornado   31568    0  1311  0.723803  22849
1        Drought   65644    0   521  0.257221  16885
2   Winter Storm  414873    0   300  0.027572  11439
3          Flood   99038    0   802  0.102587  10160
4      High Wind  319759    0   101  0.027055   8651
5      Lightning   14229    0    51  0.377961   5378
6      Hurricane    1777    0  2409  2.181204   3876
7          OTHER    9897    0    64  0.224108   2218
8       Wildfire    5251    0   104  0.403923   2121
9    Rip Current    8325    0    55  0.209730   1746
10    Heavy Rain   15217    0    52  0.037064    564

In [178]:
X1 = df1.groupby('EPISODE_ID').CASUALTIES.agg(['count','min', 'max', 'mean', 'sum']).sort_values('sum', axis=0, ascending=False).reset_index().head(11)
X1

EPISODE_ID  count  min   max        mean   sum
0      1150299     28    0   802  155.178571  4345
1        23202     25    0  2409   97.800000  2445
2        50455    119    0   852   18.067227  2150
3        49972     32    0  1311   41.093750  1315
4        10217     22    0   521   43.681818   961
5       178038      5    0   704  157.400000   787
6      1150309      4   50   500  187.500000   750
7        82875    143    0   304    5.000000   715
8        70489      4   50   250  137.500000   550
9        71079      3   74   300  158.000000   474
10       49915    165    0   204    2.763636   456

In [173]:
df1[df1.EPISODE_ID == 23202].COMBINED_DAMAGE.agg(['count','min', 'max', 'mean', 'sum', ]).sort_values('sum', axis=0, ascending=False).reset_index().head(11))

13431080000.0

df1[(df1.CASUALTIES == 2409) & (df1.EPISODE_ID == '23202')  
    (df1.CASUALTIES == 1311) & (df1.EPISODE_ID == '49972') | 
    (df1.CASUALTIES == 852) & (df1.EPISODE_ID == '50455') |
    (df1.CASUALTIES == 802) & (df1.EPISODE_ID == '1150299') | 
    (df1.CASUALTIES == 704) & (df1.EPISODE_ID == '178038') | 
    (df1.CASUALTIES == 521) & (df1.EPISODE_ID == '10217') | 
    (df1.CASUALTIES == 500) & (df1.EPISODE_ID == '1150309') | 
    (df1.CASUALTIES == 450) & (df1.EPISODE_ID == '79351')       
    (df1.CASUALTIES == 441) & (df1.EPISODE_ID == '216163') | 
    (df1.CASUALTIES == 355) & (df1.EPISODE_ID == '75828')    
   
   
   ].sort_values('CASUALTIES', axis=0, ascending=False)

In [160]:
a1 = df1.groupby('EVENT_TYPE').CASUALTIES.agg(['max']).sort_values('max', axis=0, ascending=False).reset_index().head(10)
a1

EVENT_TYPE   max
0       Hurricane  2409
1         Tornado  1311
2           Flood   802
3         Drought   521
4    Winter Storm   300
5         Tsunami   180
6  Tropical Storm   154
7        Wildfire   104
8       High Wind   101
9           OTHER    64

In [161]:
df1[(df1.CASUALTIES == 2409) & (df1.EVENT_TYPE == 'Hurricane') | 
    (df1.CASUALTIES == 1311) & (df1.EVENT_TYPE == 'Tornado') | 
    (df1.CASUALTIES == 802) & (df1.EVENT_TYPE == 'Flood') |
    (df1.CASUALTIES == 521) & (df1.EVENT_TYPE == 'Drought') | 
    (df1.CASUALTIES == 300) & (df1.EVENT_TYPE == 'Winter Storm') | 
    (df1.CASUALTIES == 180) & (df1.EVENT_TYPE == 'Tsunami') | 
    (df1.CASUALTIES == 104) & (df1.EVENT_TYPE == 'Wildfire') | 
    (df1.CASUALTIES == 101) & (df1.EVENT_TYPE == 'High Wind')       
   ].sort_values('CASUALTIES', axis=0, ascending=False)

Unnamed: 0  BEGIN_DAY  EPISODE_ID  EVENT_ID           STATE  YEAR  \
667464       60644         12       23202    136433           TEXAS  2008   
832105       33549         22       49972    296617        MISSOURI  2011   
135065       44513         17     1150299   5672496           TEXAS  1998   
588131       40469          4       10217     56732        MISSOURI  2007   
944941        2834         24       71079    431502            UTAH  2013   
726432       48068         29       35252    204396  AMERICAN SAMOA  2009   
386973       47247         25      163276   5335898      CALIFORNIA  2003   
895316       17668         28       62263    377350        MISSOURI  2012   

       MONTH_NAME    EVENT_TYPE                       CZ_NAME  WFO  \
667464  September     Hurricane                        HARRIS  HGX   
832105        May       Tornado                        JASPER  SGF   
135065  October           Flood                         COMAL  EWX   
588131     August       Drought                     ST. LOUIS  LSX   
944941    January  Winter Storm  SALT LAKE AND TOOELE VALLEYS  SLC   
726432  September       Tsunami                       TUTUILA  ASO   
386973  October        Wildfire                     SAN DIEGO  SGX   
895316      April     High Wind                 ST. LOUIS (C)  LSX   

             ...         TOR_OTHER_WFO  TOR_OTHER_CZ_STATE  END_LAT  \
667464       ...                   NaN                 NaN      NaN   
832105       ...                   SGF                  MO    37.05   
135065       ...                   NaN                 NaN      NaN   
588131       ...                   NaN                 NaN      NaN   
944941       ...                   NaN                 NaN      NaN   
726432       ...                   NaN                 NaN      NaN   
386973       ...                   NaN                 NaN      NaN   
895316       ...                   NaN                 NaN      NaN   

                                        EPISODE_NARRATIVE  \
667464  The eye of Hurricane Ike moved ashore in Galve...   
832105  A strong upper level trough across the central...   
135065  The Great October Flood\n\nIn advance of a ver...   
588131  The first and only Heat Wave of the summer sta...   
944941  A rare freezing rain event occurred in Utah on...   
726432  A tsunami wave affected all of American Samoa ...   
386973  Named the Cedar Fire, this fire was started ar...   
895316  An area of low pressure moved east along Inter...   

                                          EVENT_NARRATIVE  CASUALTIES  \
667464  Ike produced damage due to high storm surge an...        2409   
832105  National Weather Service survey teams rated th...        1311   
135065                                                NaN         802   
588131                                                NaN         521   
944941  Of all the areas across Utah, the Salt Lake Va...         300   
726432  A 7.9 earthquake occurred about 120 miles sout...         180   
386973                                                NaN         104   
895316  About 100 people were injured and one person w...         101   

       TOTAL_DEATHS  TOTAL_INJURIES        DATE  COMBINED_DAMAGE  
667464            9            2400  2008-09-12     2.000000e+09  
832105          161            1150  2011-05-22     2.800000e+09  
135065            2             800  1998-10-17     5.005000e+07  
588131            2             519  2007-08-04     0.000000e+00  
944941            0             300  2013-01-24     0.000000e+00  
726432           32             148  2009-09-29     8.102000e+07  
386973           14              90  2003-10-25     1.046500e+09  
895316            1             100  2012-04-28     0.000000e+00  

[8 rows x 31 columns]

In [147]:
c1 = df1.groupby('EVENT_TYPE').TOTAL_DEATHS.agg(['max']).sort_values('max', axis=0, ascending=False).reset_index().head(10)
c1

EVENT_TYPE  max
0         Tornado  161
1         Drought   93
2         Tsunami   32
3           Flood   23
4  Tropical Storm   22
5      Heavy Rain   19
6        Wildfire   19
7           OTHER   14
8    Winter Storm   14
9     Rip Current   12

In [149]:
df1[(df1.TOTAL_DEATHS == 161) & (df1.EVENT_TYPE == 'Tornado') | 
    (df1.TOTAL_DEATHS == 93) & (df1.EVENT_TYPE == 'Drought') | 
    (df1.TOTAL_DEATHS == 32) & (df1.EVENT_TYPE == 'Tsunami') |
    (df1.TOTAL_DEATHS == 23) & (df1.EVENT_TYPE == 'Flood') | 
    (df1.TOTAL_DEATHS == 22) & (df1.EVENT_TYPE == 'Tropical Storm') | 
    (df1.TOTAL_DEATHS == 19) & (df1.EVENT_TYPE == 'Heavy Rain') | 
    (df1.TOTAL_DEATHS == 19) & (df1.EVENT_TYPE == 'Wildfire') 
   ].sort_values('TOTAL_DEATHS', axis=0, ascending=False)

Unnamed: 0  BEGIN_DAY  EPISODE_ID  EVENT_ID           STATE  YEAR  \
832105       33549         22       49972    296617        MISSOURI  2011   
176095       34570         28       85699   5710162        ILLINOIS  1999   
726432       48068         29       35252    204396  AMERICAN SAMOA  2009   
934110       56462         29       70044    420534        NEW YORK  2012   
262547       22632          5      118293   5256253           TEXAS  2001   
63107        14546         19       50880   5624210            GUAM  1997   
976522       34415         28       76198    459593         ARIZONA  2013   

       MONTH_NAME      EVENT_TYPE                   CZ_NAME  WFO  \
832105        May         Tornado                    JASPER  SGF   
176095  July              Drought                      COOK  LOT   
726432  September         Tsunami                   TUTUILA  ASO   
934110    October           Flood     RICHMOND (STATEN IS.)  OKX   
262547  June       Tropical Storm                    HARRIS  HGX   
63107   April          Heavy Rain                MICRONESIA  GUA   
976522       June        Wildfire  YAVAPAI COUNTY MOUNTAINS  FGZ   

             ...         TOR_OTHER_WFO  TOR_OTHER_CZ_STATE  END_LAT  \
832105       ...                   SGF                  MO    37.05   
176095       ...                   NaN                 NaN      NaN   
726432       ...                   NaN                 NaN      NaN   
934110       ...                   NaN                 NaN      NaN   
262547       ...                   NaN                 NaN      NaN   
63107        ...                   NaN                 NaN      NaN   
976522       ...                   NaN                 NaN      NaN   

                                        EPISODE_NARRATIVE  \
832105  A strong upper level trough across the central...   
176095  An intense heat wave began in the Chicago metr...   
726432  A tsunami wave affected all of American Samoa ...   
934110  Post Tropical Storm Sandy was the costliest na...   
262547  Tropical Storm Allison formed in the northwest...   
63107   Typhoon Isa (02W) was within the Guam Area of ...   
976522  The Yarnell Hill Fire started around 536 PM by...   

                                          EVENT_NARRATIVE  CASUALTIES  \
832105  National Weather Service survey teams rated th...        1311   
176095                                                NaN          93   
726432  A 7.9 earthquake occurred about 120 miles sout...         180   
934110  Widespread record coastal flooding occurred in...          23   
262547                                                NaN          22   
63107                                                 NaN          27   
976522  The Yarnell Hill Fire was started by lightning...          19   

       TOTAL_DEATHS  TOTAL_INJURIES        DATE  COMBINED_DAMAGE  
832105          161            1150  2011-05-22     2.800000e+09  
176095           93               0  1999-07-28     0.000000e+00  
726432           32             148  2009-09-29     8.102000e+07  
934110           23               0  2012-10-29     0.000000e+00  
262547           22               0  2001-06-05     5.030000e+09  
63107            19               8  1997-04-19     2.000000e+06  
976522           19               0  2013-06-28     0.000000e+00  

[7 rows x 31 columns]

In [143]:
b1 = df1.groupby('EVENT_TYPE').TOTAL_INJURIES.agg(['max']).sort_values('max', axis=0, ascending=False).reset_index().head(10)
b1

EVENT_TYPE   max
0       Hurricane  2400
1         Tornado  1150
2           Flood   800
3         Drought   519
4    Winter Storm   300
5  Tropical Storm   154
6         Tsunami   148
7       High Wind   100
8        Wildfire    90
9           OTHER    62

In [148]:
df1[(df1.TOTAL_INJURIES == 2400) & (df1.EVENT_TYPE == 'Hurricane') | 
    (df1.TOTAL_INJURIES == 1150) & (df1.EVENT_TYPE == 'Tornado') | 
    (df1.TOTAL_INJURIES == 800) & (df1.EVENT_TYPE == 'Flood') |
    (df1.TOTAL_INJURIES == 519) & (df1.EVENT_TYPE == 'Drought') | 
    (df1.TOTAL_INJURIES == 300) & (df1.EVENT_TYPE == 'Winter Storm') | 
    (df1.TOTAL_INJURIES == 154) & (df1.EVENT_TYPE == 'Tropical Storm') | 
    (df1.TOTAL_INJURIES == 100) & (df1.EVENT_TYPE == 'High Wind') |
    (df1.TOTAL_INJURIES == 148) & (df1.EVENT_TYPE == 'Tsunami') | 
    (df1.TOTAL_INJURIES == 90) & (df1.EVENT_TYPE == 'Wildfire') 
   ].sort_values('TOTAL_INJURIES', axis=0, ascending=False)

Unnamed: 0  BEGIN_DAY  EPISODE_ID  EVENT_ID           STATE  YEAR  \
667464       60644         12       23202    136433           TEXAS  2008   
832105       33549         22       49972    296617        MISSOURI  2011   
135065       44513         17     1150299   5672496           TEXAS  1998   
588131       40469          4       10217     56732        MISSOURI  2007   
944941        2834         24       71079    431502            UTAH  2013   
385085       45359         18      163734   5335410        MARYLAND  2003   
726432       48068         29       35252    204396  AMERICAN SAMOA  2009   
895316       17668         28       62263    377350        MISSOURI  2012   
386973       47247         25      163276   5335898      CALIFORNIA  2003   

       MONTH_NAME      EVENT_TYPE                       CZ_NAME  WFO  \
667464  September       Hurricane                        HARRIS  HGX   
832105        May         Tornado                        JASPER  SGF   
135065  October             Flood                         COMAL  EWX   
588131     August         Drought                     ST. LOUIS  LSX   
944941    January    Winter Storm  SALT LAKE AND TOOELE VALLEYS  SLC   
385085  September  Tropical Storm                    ST. MARY'S  LWX   
726432  September         Tsunami                       TUTUILA  ASO   
895316      April       High Wind                 ST. LOUIS (C)  LSX   
386973  October          Wildfire                     SAN DIEGO  SGX   

             ...         TOR_OTHER_WFO  TOR_OTHER_CZ_STATE  END_LAT  \
667464       ...                   NaN                 NaN      NaN   
832105       ...                   SGF                  MO    37.05   
135065       ...                   NaN                 NaN      NaN   
588131       ...                   NaN                 NaN      NaN   
944941       ...                   NaN                 NaN      NaN   
385085       ...                   NaN                 NaN      NaN   
726432       ...                   NaN                 NaN      NaN   
895316       ...                   NaN                 NaN      NaN   
386973       ...                   NaN                 NaN      NaN   

                                        EPISODE_NARRATIVE  \
667464  The eye of Hurricane Ike moved ashore in Galve...   
832105  A strong upper level trough across the central...   
135065  The Great October Flood\n\nIn advance of a ver...   
588131  The first and only Heat Wave of the summer sta...   
944941  A rare freezing rain event occurred in Utah on...   
385085  On September 18, 2003, Hurricane Isabel made l...   
726432  A tsunami wave affected all of American Samoa ...   
895316  An area of low pressure moved east along Inter...   
386973  Named the Cedar Fire, this fire was started ar...   

                                          EVENT_NARRATIVE  CASUALTIES  \
667464  Ike produced damage due to high storm surge an...        2409   
832105  National Weather Service survey teams rated th...        1311   
135065                                                NaN         802   
588131                                                NaN         521   
944941  Of all the areas across Utah, the Salt Lake Va...         300   
385085                                                NaN         154   
726432  A 7.9 earthquake occurred about 120 miles sout...         180   
895316  About 100 people were injured and one person w...         101   
386973                                                NaN         104   

       TOTAL_DEATHS  TOTAL_INJURIES        DATE  COMBINED_DAMAGE  
667464            9            2400  2008-09-12     2.000000e+09  
832105          161            1150  2011-05-22     2.800000e+09  
135065            2             800  1998-10-17     5.005000e+07  
588131            2             519  2007-08-04     0.000000e+00  
944941            0             300  2013-01-24     0.000000e+00  
385085            0             154  2003-09-18     8.625000e

In [150]:
d1 = df1.groupby('EVENT_TYPE').DAMAGE_PROPERTY.agg(['max']).sort_values('max', axis=0, ascending=False).reset_index().head(10)
d1

EVENT_TYPE           max
0           Flood  1.150000e+11
1     Rip Current  1.790000e+10
2       Hurricane  1.000000e+10
3  Tropical Storm  5.030000e+09
4         Tornado  2.800000e+09
5    Winter Storm  1.800000e+09
6       High Wind  1.500000e+09
7        Wildfire  1.500000e+09
8         Drought  2.500000e+08
9         Tsunami  8.100000e+07

In [151]:
df1[(df1.DAMAGE_PROPERTY == 1.150000e+11) & (df1.EVENT_TYPE == 'Flood') | 
    (df1.DAMAGE_PROPERTY == 1.790000e+10) & (df1.EVENT_TYPE == 'Rip Current') | 
    (df1.DAMAGE_PROPERTY == 1.000000e+10) & (df1.EVENT_TYPE == 'Hurricane') |
    (df1.DAMAGE_PROPERTY == 2.800000e+09) & (df1.EVENT_TYPE == 'Tornado') | 
    (df1.DAMAGE_PROPERTY == 1.800000e+09) & (df1.EVENT_TYPE == 'Winter Storm') | 
    (df1.DAMAGE_PROPERTY == 1.500000e+09) & (df1.EVENT_TYPE == 'High Wind') | 
    (df1.DAMAGE_PROPERTY == 1.500000e+09) & (df1.EVENT_TYPE == 'WildFire') |
    (df1.DAMAGE_PROPERTY == 2.500000e+08) & (df1.EVENT_TYPE == 'Drought') | 
    (df1.DAMAGE_PROPERTY == 8.100000e+07) & (df1.EVENT_TYPE == 'Tsunami') 
   ].sort_values('DAMAGE_PROPERTY', axis=0, ascending=False)

Unnamed: 0  BEGIN_DAY  EPISODE_ID  EVENT_ID           STATE  YEAR  \
499256         189          1      203478   5486532      CALIFORNIA  2006   
486473       41382         29      198567   5475650       LOUISIANA  2005   
492055       46964         24      199954   5477628         FLORIDA  2005   
832105       33549         22       49972    296617        MISSOURI  2011   
788177       52416          5       44018    266216         ARIZONA  2010   
934142       56494         29       68471    416960      NEW JERSEY  2012   
849642       51086          1       53560    320625           TEXAS  2011   
726432       48068         29       35252    204396  AMERICAN SAMOA  2009   

       MONTH_NAME    EVENT_TYPE             CZ_NAME  WFO       ...         \
499256  January           Flood                NAPA  MTR       ...          
486473  August      Rip Current             ORLEANS  LIX       ...          
492055  October       Hurricane  COASTAL PALM BEACH  MFL       ...          
832105        May       Tornado              JASPER  SGF       ...          
788177    October  Winter Storm            MARICOPA  PSR       ...          
934142    October     High Wind    EASTERN MONMOUTH  PHI       ...          
849642       July       Drought             LUBBOCK  LUB       ...          
726432  September       Tsunami             TUTUILA  ASO       ...          

        TOR_OTHER_WFO  TOR_OTHER_CZ_STATE  END_LAT  \
499256            NaN                 NaN    38.47   
486473            NaN                 NaN      NaN   
492055            NaN                 NaN      NaN   
832105            SGF                  MO    37.05   
788177            NaN                 NaN    33.59   
934142            NaN                 NaN      NaN   
849642            NaN                 NaN      NaN   
726432            NaN                 NaN      NaN   

                                        EPISODE_NARRATIVE  \
499256  Major flooding continued into the early hours ...   
486473  Storm surge damage in southeast Louisiana, esp...   
492055  Wilma was a classic October hurricane which st...   
832105  A strong upper level trough across the central...   
788177  An unusual series of severe storms with damagi...   
934142  Post Tropical Storm Sandy was the costliest na...   
849642  Exceptional (D4) drought continued across the ...   
726432  A tsunami wave affected all of American Samoa ...   

                                          EVENT_NARRATIVE  CASUALTIES  \
499256                                                NaN           0   
486473                                                NaN           0   
492055                                                NaN           1   
832105  National Weather Service survey teams rated th...        1311   
788177  This storm moved from Firebird Lake, south of ...           1   
934142                                                NaN           0   
849642                                                NaN           0   
726432  A 7.9 earthquake occurred about 120 miles sout...         180   

       TOTAL_DEATHS  TOTAL_INJURIES        DATE  COMBINED_DAMAGE  
499256            0               0  2006-01-01     1.150325e+11  
486473            0               0  2005-08-29     1.790000e+10  
492055            1               0  2005-10-24     1.000000e+10  
832105          161            1150  2011-05-22     2.800000e+09  
788177            0               1  2010-10-05     1.800000e+09  
934142            0               0  2012-10-29     1.500000e+09  
849642            0               0  2011-07-01     5.000000e+08  
726432           32             148  2009-09-29     8.102000e+07  

[8 rows x 31 columns]